### PR TITLE
[MSHARED-763] - Include a dependency change detection.

### PR DESCRIPTION
**Pull request by LiwaeLamaa**

### CODE REVIEW

File: src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java [hunk 1]
```diff
@@ -359,6 +378,143 @@ public void afterRebuildExecution( IncrementalBuildHelperRequest incrementalBuil
...
+        {
+            //no file, assume rebuild
+            return true;
+        }
+
+        Map<File, Long> origDependencyInfo;
+        List<File> origClasspathElements;
+        try
+        {
+            ObjectInputStream ois = new ObjectInputStream ( new FileInputStream( mojoConfigFile ) );
```
Comment (olamy) [inline src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java]: we are java7 now so better using try resources or at least ois.close in finally block

File: src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java [hunk 2]
```diff
@@ -359,6 +378,143 @@ public void afterRebuildExecution( IncrementalBuildHelperRequest incrementalBuil
...
+            origClasspathElements = ( List<File> ) ois.readObject();
+            origDependencyInfo = ( Map<File, Long> ) ois.readObject();
+            ois.close();
+        }
+        catch ( FileNotFoundException e )
+        {
+            //no file, assume rebuild
+            return true;
```
Comment (olamy) [inline src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java]: at least log the exception so user knows there is a problem...

File: src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java [hunk 3]
```diff
@@ -359,6 +378,143 @@ public void afterRebuildExecution( IncrementalBuildHelperRequest incrementalBuil
...
+        }
+        catch ( IOException e )
+        {
+            //assume rebuild
+            return true;
+        }
+        catch ( ClassNotFoundException e )
+        {
```
Comment (olamy) [inline src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java]: java7 so please catch with multiple exceptions

File: src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java [hunk 4]
```diff
@@ -359,6 +378,143 @@ public void afterRebuildExecution( IncrementalBuildHelperRequest incrementalBuil
...
+
+        if ( dependencyInfo == null )
+        {
+            mojoConfigFile.delete();
+            return;
+        }
+
+        try
+        {
+            ObjectOutputStream oos = new ObjectOutputStream( new FileOutputStream( mojoConfigFile ) );
```
Comment (olamy) [inline src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java]: use try resources

**Comment by jira-importer** [comment]
Resolve #38

**Pull request closed without merging**